tests/doc/parse.ts

```diff
@@ -819,3 +819,10 @@ describe('CRLF line endings', () => {
     expect(res).toBe('foo bar')
   })
 })
+
+describe('Failed to decode suffix', () => {
+  test('rise in parse for malformed uri', () => {
+    const data = `[!!eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeJJJJJJJeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeebooleaeeeeeeeeeee-eeeeeeeee%eeeeeeeeeeeee[`
+    expect(() => YAML.parse(data)).toThrow(/Failed to decode suffix/)
```

Comment:
A few notes here:
- the test should call `YAML.parseDocument`, which should never throw.
- the check should be on the `doc.errors` contents; see for example the duplicate keys tests above.
- the source should not be triggering any other errors, so something like `!!%ee 0`.

Reply:
Yeah, that works.